[BUG]: linux: set _FILE_OFFSET_BITS=64 and use non-lfs64 calls

### Describe the Bug

(i don't know which template to write generic build issues in)  

currently in `pcsx2/Linux/LnxFlatFileReader.cpp`, there is:  

```cpp
uint FlatFileReader::GetBlockCount(void) const
{
#if defined(__HAIKU__) || defined(__APPLE__) || defined(__FreeBSD__)
	struct stat sysStatData;
	if (fstat(m_fd, &sysStatData) < 0)
		return 0;
#else
	struct stat64 sysStatData;
	if (fstat64(m_fd, &sysStatData) < 0)
		return 0;
#endif

	return (int)(sysStatData.st_size / m_blocksize);
}
```

on linux, this is using the 'lfs64 compat function defines', generally exposed by `-D_LARGEFILE_SOURCE`.

however, this in itself is mostly legacy. the modern way to do this is to have only:

```
struct stat sysStatData;
if (fstat( ...
```

and pass `-D_FILE_OFFSET_BITS=64`. this makes the default functions 64-bit types and utilises a 64-bit off_t, etc, on both 32-bit and 64-bit architectures. [small meaningless reference](https://stackoverflow.com/a/42597059).

utilising the `-D_FILE_OFFSET_BITS=64` would fix compiling against musl master, which no longer exposes the `*64` apis by default with `-D_GNU_SOURCE`, intentionally: https://github.com/bminor/musl/commit/25e6fee27f4a293728dd15b659170e7b9c7db9bc  

(aside: musl does not have 32-bit interfaces for this to begin with, off_t is always 64 regardless, these are just compat shims, but this doesn't matter here)  

it would also be portable to all other linux libcs- it's required to respect `-D_FILE_OFFSET_BITS=64` to expose the 64-bit-accepting interfaces for these non-64-prefixed functions.


tl;dr:

- change `stat64/fstat64` to just `stat/fstat`
- define `-D_FILE_OFFSET_BITS=64` when building on linux

i would submit an mr but don't exactly know where you would want to add this in cmake.

### Reproduction Steps

build pcsx2 against musl-libc master without a deprecated `-D_LARGEFILE64_SOURCE` macro, and observe:  


```
/home/demon/src/aports/testing/pcsx2/src/pcsx2-1.7.4231/pcsx2/Linux/LnxFlatFileReader.cpp:108:16: error: variable has incomplete type 'struct stat64'
        struct stat64 sysStatData;
                      ^
```

(type doesn't exist)

### Expected Behavior

_No response_

### PCSX2 Revision

v1.7.4231

### Operating System

Linux (64bit) - Specify distro below

### If Linux - Specify Distro

alpine

--- a/common/FileSystem.cpp
+++ b/common/FileSystem.cpp
@@ -726,13 +726,6 @@ int FileSystem::FSeek64(std::FILE* fp, s64 offset, int whence)
 #ifdef _WIN32
 	return _fseeki64(fp, offset, whence);
 #else
-	// Prevent truncation on platforms which don't have a 64-bit off_t.
-	if constexpr (sizeof(off_t) != sizeof(s64))
-	{
-		if (offset < std::numeric_limits<off_t>::min() || offset > std::numeric_limits<off_t>::max())
-			return -1;
-	}
-
 	return fseeko(fp, static_cast<off_t>(offset), whence);
 #endif
 }
@@ -1474,6 +1467,9 @@ bool FileSystem::SetPathCompression(const char* path, bool enable)
 
 #else
 
+// No 32-bit file offsets breaking stuff please.
+static_assert(sizeof(off_t) == sizeof(s64));
+
 static u32 RecursiveFindFiles(const char* OriginPath, const char* ParentPath, const char* Path, const char* Pattern,
 	u32 Flags, FileSystem::FindResultsArray* pResults)
 {
@@ -1528,16 +1524,9 @@ static u32 RecursiveFindFiles(const char* OriginPath, const char* ParentPath, co
 		FILESYSTEM_FIND_DATA outData;
 		outData.Attributes = 0;
 
-#if defined(__HAIKU__) || defined(__APPLE__) || defined(__FreeBSD__)
 		struct stat sDir;
 		if (stat(full_path.c_str(), &sDir) < 0)
 			continue;
-
-#else
-		struct stat64 sDir;
-		if (stat64(full_path.c_str(), &sDir) < 0)
-			continue;
-#endif
 
 		if (S_ISDIR(sDir.st_mode))
 		{
@@ -1640,14 +1629,9 @@ bool FileSystem::StatFile(const char* path, FILESYSTEM_STAT_DATA* sd)
 	if (path[0] == '\0')
 		return false;
 
-		// stat file
-#if defined(__HAIKU__) || defined(__APPLE__) || defined(__FreeBSD__)
+	// stat file
 	struct stat sysStatData;
 	if (stat(path, &sysStatData) < 0)
-#else
-	struct stat64 sysStatData;
-	if (stat64(path, &sysStatData) < 0)
-#endif
 		return false;
 
 	// parse attributes
@@ -1673,14 +1657,9 @@ bool FileSystem::StatFile(std::FILE* fp, FILESYSTEM_STAT_DATA* sd)
 	if (fd < 0)
 		return false;
 
-		// stat file
-#if defined(__HAIKU__) || defined(__APPLE__) || defined(__FreeBSD__)
+	// stat file
 	struct stat sysStatData;
 	if (fstat(fd, &sysStatData) < 0)
-#else
-	struct stat64 sysStatData;
-	if (fstat64(fd, &sysStatData) < 0)
-#endif
 		return false;
 
 	// parse attributes
@@ -1706,14 +1685,9 @@ bool FileSystem::FileExists(const char* path)
 	if (path[0] == '\0')
 		return false;
 
-		// stat file
-#if defined(__HAIKU__) || defined(__APPLE__) || defined(__FreeBSD__)
+	// stat file
 	struct stat sysStatData;
 	if (stat(path, &sysStatData) < 0)
-#else
-	struct stat64 sysStatData;
-	if (stat64(path, &sysStatData) < 0)
-#endif
 		return false;
 
 	if (S_ISDIR(sysStatData.st_mode))
@@ -1728,14 +1702,9 @@ bool FileSystem::DirectoryExists(const char* path)
 	if (path[0] == '\0')
 		return false;
 
-		// stat file
-#if defined(__HAIKU__) || defined(__APPLE__) || defined(__FreeBSD__)
+	// stat file
 	struct stat sysStatData;
 	if (stat(path, &sysStatData) < 0)
-#else
-	struct stat64 sysStatData;
-	if (stat64(path, &sysStatData) < 0)
-#endif
 		return false;
 
 	if (S_ISDIR(sysStatData.st_mode))

--- a/common/Linux/LnxHostSys.cpp
+++ b/common/Linux/LnxHostSys.cpp
@@ -250,13 +250,9 @@ void* HostSys::CreateSharedMemory(const char* name, size_t size)
 	shm_unlink(name);
 
 	// ensure it's the correct size
-#if !defined(__APPLE__) && !defined(__FreeBSD__)
-	if (ftruncate64(fd, static_cast<off64_t>(size)) < 0)
-#else
 	if (ftruncate(fd, static_cast<off_t>(size)) < 0)
-#endif
 	{
-		std::fprintf(stderr, "ftruncate64(%zu) failed: %d\n", size, errno);
+		std::fprintf(stderr, "ftruncate(%zu) failed: %d\n", size, errno);
 		return nullptr;
 	}
 

--- a/pcsx2/Linux/LnxFlatFileReader.cpp
+++ b/pcsx2/Linux/LnxFlatFileReader.cpp
@@ -88,7 +88,6 @@ void FlatFileReader::CancelRead(void)
 
 void FlatFileReader::Close(void)
 {
-
 	if (m_fd != -1)
 		close(m_fd);
 
@@ -100,15 +99,9 @@ void FlatFileReader::Close(void)
 
 uint FlatFileReader::GetBlockCount(void) const
 {
-#if defined(__HAIKU__) || defined(__APPLE__) || defined(__FreeBSD__)
 	struct stat sysStatData;
 	if (fstat(m_fd, &sysStatData) < 0)
 		return 0;
-#else
-	struct stat64 sysStatData;
-	if (fstat64(m_fd, &sysStatData) < 0)
-		return 0;
-#endif
 
-	return (int)(sysStatData.st_size / m_blocksize);
+	return static_cast<uint>(sysStatData.st_size / m_blocksize);
 }

